v5: Adjust docs sidebar and navbar

mdo opened the pull request:
Some tweaks to ideally improve navigating across our documentation on narrow devices. This PR removes the navbar scroll that I had implemented and replaces it with our classic collapsing navbar, but with a slight twist. I've added column classes to stack the nav across two columns on mobile devices so that the opened nav isn't too tall.

Still need to adjust the menu toggles (missing some styles, tap targets could be larger, etc), but hoping to get some initial feedback from the team on overall direction. Also wondering if I should do more to condense the subnav (search, version picker, and sidebar toggle) on mobile views.

Any thoughts, feedback, or examples of other patterns would be appreciated!

/cc @twbs/css-review 

Preview: https://deploy-preview-31119--twbs-bootstrap.netlify.app/

mdo committed:
Hide the docs nav on mobile, redesign it for two columns on narrow devices

mdo committed:
Tweak spacing on ToC

mdo committed:
Redesign sidebar and subnav

patrickhlauke commented:
like this, particularly since small screen doesn't necessarily always mean "mobile user with touch" (it could, for instance, still be a mouse user on desktop, but with high magnification/zoom). and the horizontal scrolling without scrollbar is unusable for these users, currently.

alecpl commented:
I'd like search/version/toggle in a single line. Maybe the version selector could be visually integrated with the search input (and does not need the "Bootstrap" label, only version number). Maybe the icon should be different than the main menu icon (folder tree, hierarchy?).

btw, on the home page (not docs, the main page) I propose to make `.masthead-followup-icon` in the same line with the following `h2`. And smaller bottom padding/margin on `#content`.

mdo committed:
Revamp subnav mobile layout so everything's on one line

mdo committed:
Tighten up masthead a little

mdo commented:
Rebased to bring this up to speed with the latest `main` changes, added @alecpl's suggestion for subnav layout all on one line with a cheeky `d-none` trick, and tweaked the homepage masthead `padding` (more to come on this later I think).

Ready for review!

ffoodd commented:
I like it!

I agree with @alecpl about the two icons, there's so much lines stacking here… Not sure about a solution though, I just noticed that felt strange.

mdo committed:
Switch the icon for the subnav and do a little icon toggling

mdo commented:
Yeah the stacked bars is a little odd. How about this? Just pushed a commit for it—icon flips between the expand/collapse icons from Bootstrap Icons.

<img width="573" alt="Screen Shot 2020-09-16 at 11 20 20 AM" src="https://user-images.githubusercontent.com/98681/93376570-a7670a80-f80e-11ea-902b-febfd5b6ed28.png">

ffoodd commented:
@mdo Nice, IMHO much better!

mdo commented:
Rebased and conflicts resolved.

Ready for review!

MartijnCuppens reviewed (approved):
Neat improvement!

mdo committed:
Fix mobile overflow since we changed guters stuff

mdo committed:
Add the widths

mdo committed:
make the subnav icon purple

mdo commented:
Thanks for the reviews fam! This will be a nice little addition to Alpha 2 :D.
